### DISCUSSION TITLE
Viewbag

### DISCUSSION THREAD

**Discussion by eger-heed** [OFFLINE-GmbH/oc-boxes-support#87] [Help]
I'm using Rainlab.StaticPages for menu generation. I'm currently working on a child page and need a way to set the viewbag parameter activeMenuItem. This will be necessary on several child pages. Is there a way to set the view bag variable without creating pages in the theme folder and calling the boxesComponent on each page manually?

**Comment by tobias-kuendig** [comment]
What do you want to put in your viewBag? You can access the `boxesPage` variable on the page. Using the `cms.page.beforeRenderPage` event could work here: https://octobercms.com/docs/api/cms/page/beforerenderpage

Check if your require data is on the `$page` variable and set your view bag accordingly.
